app/assets/javascripts/modal/modal-fetch.js

```diff
@@ -23,7 +23,10 @@ window.ModalFetch.getLink = function (item) {
 
 window.ModalFetch.postForm = function (form) {
   var controller = new window.AbortController()
-  setTimeout(function () { controller.abort() }, 10000)
+
+  if (window.ModalFetch._notFileUpload(form)) {
```

Comment:
This is looking better, but I think it would be more idiomatic to use '!' in the if statement, rather than 'not' in the method name. We could say 'isFileUpload' instead.

Reply:
We might also want to consider using the same timeout for 'getLink' :-).

Reply:
Sure, I originally wrote it with the '!' statement but thought it may not be that obvious when it's prefixed onto the long method call.

Reply:
And sure i'll increase the timeout for getLink too!